nifi-commons/nifi-vault-utils/src/main/java/org/apache/nifi/vault/StandardVaultCommunicationService.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.vault.client.ClientHttpRequestFactoryFactory;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.Ciphertext;
+import org.springframework.vault.support.ClientOptions;
+import org.springframework.vault.support.Plaintext;
+import org.springframework.vault.support.SslConfiguration;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implements the VaultCommunicationService using Spring Vault
+ */
+public class StandardVaultCommunicationService implements VaultCommunicationService {
+    private static final String HTTPS = "https";
+
+    private final VaultConfiguration vaultConfiguration;
+    private final VaultTemplate vaultTemplate;
+
+    /**
+     * Creates a VaultCommunicationService that uses Spring Vault.
+     * @param vaultProperties Properties to configure the service
+     * @throws VaultConfigurationException If the configuration was invalid
+     */
+    public StandardVaultCommunicationService(final VaultProperties vaultProperties) throws VaultConfigurationException {
+        this.vaultConfiguration = new VaultConfiguration(vaultProperties);
+
+        final SslConfiguration sslConfiguration = vaultProperties.getUri().contains(HTTPS)
+                ? vaultConfiguration.sslConfiguration() : SslConfiguration.unconfigured();
+
+        final ClientOptions clientOptions = getClientOptions(vaultProperties);
+
+        vaultTemplate = new VaultTemplate(vaultConfiguration.vaultEndpoint(),
+                ClientHttpRequestFactoryFactory.create(clientOptions, sslConfiguration),
+                new SimpleSessionManager(vaultConfiguration.clientAuthentication()));
+    }
+
+    private static ClientOptions getClientOptions(VaultProperties vaultProperties) {
+        final ClientOptions clientOptions = new ClientOptions();
+        Duration readTimeoutDuration = clientOptions.getReadTimeout();
+        Duration connectionTimeoutDuration = clientOptions.getConnectionTimeout();
+        final Optional<String> configuredReadTimeout = vaultProperties.getReadTimeout();
+        if (configuredReadTimeout.isPresent()) {
+            readTimeoutDuration = getDuration(configuredReadTimeout.get());
+        }
+        final Optional<String> configuredConnectionTimeout = vaultProperties.getConnectionTimeout();
+        if (configuredConnectionTimeout.isPresent()) {
+            connectionTimeoutDuration = getDuration(configuredConnectionTimeout.get());
+        }
+        return new ClientOptions(connectionTimeoutDuration, readTimeoutDuration);
+    }
+
+    private static Duration getDuration(String formattedDuration) {
+        final double duration = FormatUtils.getPreciseTimeDuration(formattedDuration, TimeUnit.MILLISECONDS);
+        return Duration.ofMillis(Double.valueOf(duration).longValue());
+    }
+
+    @Override
+    public String encrypt(String transitKey, byte[] plainText) {
```

Comment:
Recommend marking these method parameters and others as `final`.
```suggestion
    public String encrypt(final String transitKey, final byte[] plainText) {
```